lib/active_merchant/billing/gateways/flex_charge.rb

```diff
@@ -228,8 +254,8 @@ def message_from(response)
         response[:title] || response[:responseMessage] || response[:status]
       end
 
-      def authorization_from(response)
-        response[:orderSessionKey]
+      def authorization_from(action, response)
+        action == :store ? response[:transaction][:payment_method][:token] : response[:orderSessionKey]
```

Comment:
💬 Comment:

Just in case of unexpected scenarios `dig` is a safer way to get the token